### PR TITLE
Install theme when clicking the image

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes: #496

### CODE REVIEW

File: tests/client/disco/components/TestAddon.js [hunk 1]
```diff
@@ -224,10 +225,28 @@ describe('<Addon />', () => {
       assert.ok(themeAction.calledWith(themeImage, THEME_RESET_PREVIEW));
     });
 
-    it('runs preventDefault onClick', () => {
+    it('installs a theme when the theme image is clicked', () => {
       const preventDefault = sinon.stub();
-      Simulate.click(themeImage, {preventDefault});
+      const installTheme = sinon.stub();
+      const data = { ...result, type: THEME_TYPE, themeAction,
```
Comment (mstriemer) [inline tests/client/disco/components/TestAddon.js]: I don't think there should be spaces in the parens here. I find the indentation a bit weird, too.

File: tests/client/disco/components/TestAddon.js [hunk 2]
```diff
@@ -224,10 +225,28 @@ describe('<Addon />', () => {
...
+      themeImage = findDOMNode(root).querySelector('.theme-image');
+      Simulate.click(themeImage, {currentTarget: themeImage, preventDefault});
+      assert.ok(preventDefault.called);
+      assert.ok(installTheme.called);
+    });
+
+    it('does not try to isntall theme if not UNINSTALLED', () => {
+      const preventDefault = sinon.stub();
+      const installTheme = sinon.stub();
+      const data = { ...result, type: THEME_TYPE, themeAction,
```
Comment (mstriemer) [inline tests/client/disco/components/TestAddon.js]: Same parens stuff here.

File: src/disco/components/Addon.js [hunk 3]
```diff
@@ -113,11 +114,13 @@ export class Addon extends React.Component {
...
                  data-browsertheme={this.getBrowserThemeData()}
                  onBlur={this.resetPreviewTheme}
-                 onClick={this.handleClick}
+                 onClick={this.clickInstallTheme}
                  onFocus={this.previewTheme}
                  onMouseOut={this.resetPreviewTheme}
                  onMouseOver={this.previewTheme}>
-        <img src={previewURL} alt={sprintf(i18n.gettext('Preview %(name)s'), {name})} /></a>);
+        <img src={previewURL}
+          alt={sprintf(i18n.gettext('Hover to preview or click to install %(name)s'), {name})}
```
Comment (mstriemer) [inline src/disco/components/Addon.js]: I don't think we have indentation rules in eslint but I feel like the `alt` should line up with `src` or `src` should drop down to match `alt`.

**Comment by mstriemer** [comment]
r+wc

I'm not totally sold on the idea, though. It feels weird that the thing gets focused when you click it, but de-focusing it keeps the theme around. Removing the theme when clearing focus could be even weirder though I'd imagine. Have you run this past @pwalm?

File: tests/client/disco/components/TestAddon.js (hunk 1 above)
Comment (muffinresearch) [inline tests/client/disco/components/TestAddon.js]: A little while back [1] it came up in review to add spaces when using destructuring. I guess this clearly shows we need to define when and when we aren't formatting objects with spaces and presumably if this is only in certain conditions e.g. was that first case just for a signature?

I'm going to take a look at what eslint can do for us here.

[1] https://github.com/mozilla/addons-frontend/pull/513#discussion_r65728712

**Comment by muffinresearch** [comment]
> I'm not totally sold on the idea, though. It feels weird that the thing gets focused when you click it, but de-focusing it keeps the theme around. Removing the theme when clearing focus could be even weirder though I'd imagine. Have you run this past @pwalm?

I'd be happy to get UX input on this before moving forward.  

After this I was looking at mobile, and I noticed that the mobile UX has tap for preview since there's no concept of hover. Maybe we should consider more making preview on-click then the mobile experience would be closer to this. I think that would be better and would avoid all the flashing of preview as you move the mouse over the page. I personally think things that do stuff on hover and focus aren't that nice.

File: tests/client/disco/components/TestAddon.js (hunk 1 above)
Comment (mstriemer) [inline tests/client/disco/components/TestAddon.js]: I think the difference should be: spaces if it's creating variable names, no spaces if it is creating an object. So:

```js
const { foo, baz } = {foo: 'FOO', bar: 'BAR', baz: 'BAZ'};

Comment (mstriemer) [inline tests/client/disco/components/TestAddon.js]: We should look into automating this though.

**Comment by muffinresearch** [comment]
I discussed this with @designakt and he was in favour of click to install but we should add tracking to see how much it's used. I'll update this patch to fix the conflicts and add a separate bug for adding tracking.

**Comment by mstriemer** [comment]
r+

**Pull request merged**